src/ui/common/src/components/integrations/dialogs/awsDialog.tsx

```diff
@@ -111,6 +128,99 @@ export const AWSDialog: React.FC<Props> = ({
     </Box>
   );
 
+  const k8sConfigTab = (
+    <Box>
+      <Typography variant="body2" color="gray.700">
+        Optionally configure dynamic Kubernetes cluster parameters.
+      </Typography>
+      <IntegrationTextInputField
+        spellCheck={false}
+        required={false}
+        label="Keepalive period"
+        description="How long (in seconds) does the cluster need to remain idle before it is deleted."
+        placeholder={K8sPlaceholders.keepalive}
+        onChange={(event) => {
+          k8sConfigs['keepalive'] = event.target.value;
+          onUpdateField('k8s_serialized', JSON.stringify(k8sConfigs));
+        }}
+        value={k8sConfigs['keepalive'] ?? null}
+      />
+      <IntegrationTextInputField
+        spellCheck={false}
+        required={false}
+        label="CPU node type"
+        description="The EC2 instance type of the CPU node group."
+        placeholder={K8sPlaceholders.cpu_node_type}
+        onChange={(event) => {
+          k8sConfigs['cpu_node_type'] = event.target.value;
+          onUpdateField('k8s_serialized', JSON.stringify(k8sConfigs));
```

Comment:
did you verify if this is working smoothly? It's different from what we did in some other places (where we use an additional internal state to track), but I also don't see any issue with your way

Try to type fast, delete fast, and switch between various fields. See if there's any latency when you type, and see if the created integration's fields matches what's in the dialog box. If that works, we can go with this solution.

Reply:
I think so yeah it's working.